File: 프로그래머스/unrated/120880. 특이한 정렬/특이한 정렬.py
def solution(numlist, n):
    #정렬기준을 정의하는 함수
    def sort_key(value):
        distance = abs(value - n) #n으로부터 거리계산
        #거리와 -value를 반환하여 거리가 작은 순서대로, 거리가 같으면 큰숫자가 반환
        return (distance, -value)
        
    return sorted(numlist, key=sort_key)

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.java
import java.util.Arrays;
class Solution {
    public int[] solution(int[] numlist, int n) {
        Arrays.sort(numlist);
        for(int i=0;i<numlist.length;i++){
            for(int j=0;j<numlist.length;j++){
                if(Math.abs(n-numlist[i])<=Math.abs(n-numlist[j])){
                    int temp = numlist[i];
                    numlist[i]=numlist[j];
                    numlist[j]=temp;
                }
            }
        }
        return numlist;
    }
}